nntrainer/src/neuralnet.cpp

```diff
@@ -824,6 +824,24 @@ int NeuralNetwork::addLayer(std::shared_ptr<Layer> layer) {
   return status;
 }
 
+int NeuralNetwork::addLayer_ops(std::shared_ptr<Layer> layer) {
+  int status = ML_ERROR_NONE;
+
+  LayerType type = layer->getType();
+  if (type == LAYER_UNKNOWN)
+    return ML_ERROR_INVALID_PARAMETER;
+
+  if (initialized) {
+    return ML_ERROR_NOT_SUPPORTED;
+  }
+
+  ensureName(layer);
+  std::shared_ptr<TensorOp> op = std::dynamic_pointer_cast<TensorOp>(layer);
+  graph.registerTensorOp(*op);
```

Comment:
I'm not sure if tensorop could be tensor calculation level not a layer level. Cause there are calculation with tensor values like kernel calculation or make proper data type in convolution.

Reply:
As you said, custom op for tensor is not good.
As you might have noticed, `tensor_op`'s method `compute_op` and other methods do not take any arguments.
The reason is to allow multiple arguments.

When tensorOp does tensor level calculation later, lazy tensor would have to be deprecated as tensor would do all the lazily itself.

Please tell me if I misunderstood your question.